Submitted By:            Randy McMurchy <randy_at_linuxfromscratch_dot_org>
Date:                    2004-10-31
Initial Package Version: 2.4.1
Upstream Status:         Not submitted (too much of a hack)
Origin:                  http://lists.gnu.org/archive/html/bug-gnu-utils/2003-01/msg00207.html
Description:             Fixes C++ error using GCC >= 3.x.x
                         Note: This patch is not meant to be used directly. It
                         is called by libplotter/Makefile which is created
                         with a patched Makefile.in. You'll also need:
                         plotutils-2.4.1-libplotter_Makefile-1.patch
                         plotutils-2.4.1-n_write_cc_fix-1.patch
                         plotutils-2.4.1-g_write_cc_fix-1.patch
                         plotutils-2.4.1-i_rle_cc_fix-1.patch

diff -Naur ../../plotutils-2.4.1-orig/libplot/z_write.c ./z_write.cc
--- ../../plotutils-2.4.1-orig/libplot/z_write.c        2000-06-20 04:34:42.000000000 +0000
+++ ./z_write.cc        2004-10-31 18:43:51.203482856 +0000
@@ -484,7 +484,7 @@
   ostream *stream;

   stream = (ostream *)png_get_io_ptr (png_ptr);
-  stream->write (data, length);
+  stream->write (reinterpret_cast<const char*>(data), length);
 }

 static void

